src/ophys_etl/transforms/suite2p_wrapper.py

```diff
@@ -0,0 +1,219 @@
+
+    @mm.post_load
+    def check_args(self, data, **kwargs):
+        if ('nbinned' not in data) & ('bin_size' not in data):
+            raise Suite2PWrapperException(
+                    "must provide either `nbinned` or `bin_size`")
+        return data
+
+    @mm.post_load
+    def check_retain_files(self, data, **kwargs):
+        if 'all' in data['retain_files']:
+            data['retain_files'] = [
+                    'ops1.npy', 'data.bin', 'Fneu.npy', 'F.npy', 'iscell.npy',
+                    'ops.npy', 'spks.npy', 'stat.npy']
+        return data
+
+
+class Suite2PWrapperOutputSchema(argschema.schemas.DefaultSchema):
+    output_files = argschema.fields.Dict(
+        keys=argschema.fields.Str,
+        values=argschema.fields.OutputFile,
+        required=True,
+        description="retained output files from Suite2P")
+
+
+def copy_and_add_uid(srcdir: pathlib.Path, dstdir: pathlib.Path,
+                     basenames: List[str], uid: str = None) -> List[str]:
+    """copy files matching basenames from a tree search of srcdir to
+    dstdir with an optional unique id inserted into the basename.
+
+    Parameters
+    ----------
+    srcdir : pathlib.Path
+       source directory
+    dstdir : pathlib.Path
+       destination directory
+    basenames : list
+        list of basenames to copy
+    uid : str
+        uid to insert into basename (example a timestamp string)
+
+    Returns
+    -------
+    copied_files : dict
+        keys are basenames and vaues are output paths as strings
+
+    """
+
+    copied_files = {}
+
+    for basename in basenames:
+        result = list(srcdir.rglob(basename))
+        if len(result) != 1:
+            raise ValueError(f"{len(result)} matches found in {srcdir} "
+                             f"for {basename}. Expected 1 match.")
+        dstbasename = result[0].name
+        if uid is not None:
+            dstbasename = result[0].stem + f"_{uid}" + result[0].suffix
+
+        dstfile = dstdir / dstbasename
+
+        shutil.copyfile(result[0], dstfile)
+
+        copied_files[basename] = str(dstfile)
+
+    return copied_files
+
+
+class Suite2PWrapper(argschema.ArgSchemaParser):
+    default_schema = Suite2PWrapperSchema
+    default_output_schema = Suite2PWrapperOutputSchema
+
+    def run(self):
+        self.logger.name = type(self).__name__
+        self.logger.setLevel(self.args.pop('log_level'))
+
+        # determine nbinned from bin_size
+        if 'nbinned' not in self.args:
+            with h5py.File(self.args['h5py'], 'r') as f:
+                nframes = f['data'].shape[0]
+            self.args['nbinned'] = int(nframes / self.args['bin_size'])
+            self.logger.info(f"movie has {nframes} frames. Setting nbinned "
+                             f"to {self.args['nbinned']}")
+
+        # make a tempdir for Suite2P's output
+        with tempfile.TemporaryDirectory() as tdir:
+            self.args['save_path0'] = tdir
+            suite2p.run_s2p.run_s2p(self.args)
```

Comment:
I would make a logger.info to state s2p is running and one after to state completion

Reply:
done